src/Components/Modal.js

```diff
@@ -3,54 +3,75 @@ import ReactDOM from 'react-dom';
 
 import '../styles/Modal.css'
 
-const Modal = ({
-  title,
-  poster,
-  backdrop,
-  averageRating,
-  releaseDate,
-  overview,
-  genres,
-  budget, 
-  revenue, 
-  tagline,
-  id,
-  selectedMovie,
-  isModalOpen,
-  toggleModal
-}) => {
-  const myStyle = {
-    backgroundImage: `url(${backdrop})`
+import { fetchSingleMovie } from '../api-Calls'
+
+class Modal extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      movie: this.props.selectedMovie,
+      title: null,
+      poster: null,
+      backdrop: null,
+      averageRating: null,
+      releaseDate: null,
+      overview: null,
+      genres: null,
+      budget: null,
+      revenue: null,
+      tagline: null,
+      isModalLoading: true,
+    }
   }
 
-  if (isModalOpen && selectedMovie === id) {
-    return ReactDOM.createPortal(
+
+componentDidMount = () => {
+  fetchSingleMovie(this.state.movie)
+  .then((data) => this.setState({
+    title: data.movie.title,
```

Comment:
While the way this is set up totally works, I have to ask: why not just save `data.movie` to one state property? Why break it down so far? I'm asking instead of telling you not to do this because you might have a good reason - but if it's arbitrary, I always vote for less code given it's readable. Accessing the data as `this.state.movie.budget` (etc) in the template seems readable and reasonable.

Reply:
This is a completely reasonable and excellent point Cass! We've refactored our state for Modal to only have a movie property since that helped us cut down on a significant amount of redundant code.